Migrate Manage Reusable Block to Playwright

pooja-muchandikar opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->
## What?
<!-- In a few words, what is the PR actually doing? -->
Part of https://github.com/WordPress/gutenberg/issues/38851.
Migrate[ manage-reusable-blocks.test.js ](https://github.com/WordPress/gutenberg/blob/trunk/packages/e2e-tests/specs/editor/various/manage-reusable-blocks.test.js) to its Playwright version.
## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->
Part of https://github.com/WordPress/gutenberg/issues/38851.
## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->
See[ MIGRATION.md](https://github.com/WordPress/gutenberg/blob/trunk/test/e2e/MIGRATION.md) for migration steps.
## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a Post or Page. -->
<!-- 2. Insert a Heading Block. -->
<!-- 3. etc. -->
Run` npm run test:e2e:playwright test/e2e/specs/editor/various/manage-reusable-blocks.spec.js`

pooja-muchandikar committed:
Migrate Manage Reusable Block to Playwright

pooja-muchandikar commented:
Hi @Mamaduka, 

I hope you are doing well!

I have added explicit wait on[ Line 50](https://github.com/WordPress/gutenberg/blob/9ba456f6a4ce8c8440049672c2c5b87395e58f91/test/e2e/specs/editor/various/manage-reusable-blocks.test.js#L50), the reason for this is that the test execution was too fast and further tests were failing, due to that I have added this wait.. I tried in local with other selectors but it didn't help. 

Let me know if you have any suggestions on same..

Thanks!

kevin940726 commented on test/e2e/specs/editor/various/manage-reusable-blocks.test.js:
```diff
@@ -0,0 +1,60 @@
+		const testReusableBlockFile = path.join(
+			__dirname,
+			'..',
+
```
Redundant line?

```suggestion
```

kevin940726 commented on test/e2e/specs/editor/various/manage-reusable-blocks.test.js:
```diff
@@ -0,0 +1,60 @@
+		await page.click( 'role=button[name="Import"i]' );
+
+		// Wait for the success notice.
+		await expect(
+			page.locator(
+				"div[class='notice notice-success is-dismissible'] p"
+			)
+		).toHaveText( 'Reusable block imported successfully!' );
```
Let's just wait for the text to be visible to avoid relying on CSS selectors.

```suggestion
		await expect(
			page.locator( 'text=Reusable block imported successfully!' )
		).toBeVisible();
```

kevin940726 commented on test/e2e/specs/editor/various/manage-reusable-blocks.test.js:
```diff
@@ -0,0 +1,60 @@
+		await page.waitForTimeout( 2000 );
+
+		// Refresh the page.
+		await admin.visitAdminPage( 'edit.php', 'post_type=wp_block' );
```
Would simply reloading work here?

```suggestion
		await page.reload();
```

kevin940726 commented on test/e2e/specs/editor/various/manage-reusable-blocks.test.js:
```diff
@@ -0,0 +1,60 @@
+		// Refresh the page.
+		await admin.visitAdminPage( 'edit.php', 'post_type=wp_block' );
+
+		const expectedEntries = originalEntries + 1;
+		const actualEntries = await page.locator( '.hentry' ).count();
+
+		expect( actualEntries ).toBe( expectedEntries );
```
```suggestion
		await expect( page.locator( '.hentry' ) ).toHaveCount( originalEntries + 1 );
```

kevin940726 commented on test/e2e/specs/editor/various/manage-reusable-blocks.test.js:
```diff
@@ -0,0 +1,60 @@
+	} );
+
+	test( 'Should import reusable blocks', async ( { admin, page } ) => {
+		async function getNumberOfEntries() {
+			return page.evaluate(
+				() => document.querySelectorAll( '.hentry' ).length
+			);
+		}
+
+		const originalEntries = await getNumberOfEntries();
```
Bummer that there isn't a good enough accessible selector for the entry. 😞 

```suggestion
		const originalEntries = await page.locator( '.hentry' ).count();
```

kevin940726 commented on test/e2e/specs/editor/various/manage-reusable-blocks.test.js:
```diff
@@ -0,0 +1,60 @@
+			)
+		).toHaveText( 'Reusable block imported successfully!' );
+
+		await page.waitForTimeout( 2000 );
```
Let's try removing this again. If it fails again, we can investigate why with the debugger.

kevin940726 reviewed (commented):
Thanks for your patience! Nice work!

pooja-muchandikar committed:
Address Feedbacks

pooja-muchandikar committed:
Fix Style Lint Issue

pooja-muchandikar commented:
Hi @kevin940726,

I hope you are doing great!

I have addressed all the feedbacks and pushed the commit. Please do review and let me know if there are any further feedbacks. 

Thanks!

kevin940726 reviewed (approved):
LGTM 👍  Thanks a lot!
